portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -115,12 +118,8 @@ private int getLimit(HttpServletRequest request){
          System.err.println("Could not convert to int: " + limitString);
          return -1;
      }
-     if(limit < 0){
-        System.err.println("Player choice is out of range: " + limitString);
-        return -1;
-     }
-     return limit;
```

Comment:
I think you meant to keep this line. That being said, you could probably just skip the entire variable "limit" and just return Integer.parseInt() in the if case (it can still fail and the exception will be handled as expected).